src/Workspaces/Core/Portable/Shared/Extensions/IPropertySymbolExtensions.cs

```diff
@@ -69,5 +69,19 @@ public static IPropertySymbol RenameParameters(this IPropertySymbol property, IL
                 property.SetMethod,
                 property.IsIndexer);
         }
+
+        public static bool IsWritableInConstructor(this IPropertySymbol property)
+        {
+            return property.Parameters.Length == 0 &&
+                   (property.SetMethod != null || ContainsBackingField(property));
+        }
+
+        public static IFieldSymbol GetBackingFieldIfAny(this IPropertySymbol property) =>
```

Comment:
Like this?

```
public static IFieldSymbol GetBackingFieldIfAny(this IPropertySymbol property)
    => property.ContainingType.GetMembers()
        .OfType<IFieldSymbol>()
        .FirstOrDefault(f => property.Equals(f.AssociatedSymbol));
```

---
In reply to: [86491171](https://github.com/dotnet/roslyn/pull/14827#discussion_r86491171) [](ancestors = 86491171)